Foundation/NSURLSession/NSURLSession.swift

```diff
@@ -226,6 +227,7 @@ open class URLSession : NSObject {
         self.workQueue = DispatchQueue(label: "URLSession<\(identifier)>")
         self.taskAttributesIsolation = DispatchQueue(label: "URLSession<\(identifier)>.taskAttributes", attributes: DispatchQueue.Attributes.concurrent)
         self.delegateQueue = queue ?? OperationQueue()
+        self.delegateQueue.maxConcurrentOperationCount = 1
```

Comment:
The spec. says "The queue should be a serial queue, in order to ensure the correct ordering of callbacks". But yes  it is not mandatory and user can send in a concurrent queue as well. Tested with a small test-case on Darwin and I see that user's OperationQueue is not converted to serial if it isn't already. Thanks @ianpartridge for pointing this out. 
Modified the changes for the same.